python/startup/bootstrap.py

```diff
@@ -47,36 +98,18 @@ def bootstrap_sgtk():
         error("Shotgun: Unknown platform - cannot setup ssl")
         return
 
-    try:
-        import sgtk
-    except Exception, e:
-        error("Shotgun: Could not import sgtk! Disabling for now: %s" % e)
-        return
-
-    if not "TANK_ENGINE" in os.environ:
-        error("Shotgun: Missing required environment variable TANK_ENGINE.")
-        return
-
-    engine_name = os.environ.get("TANK_ENGINE")
-    try:
-        context = sgtk.context.deserialize(os.environ.get("TANK_CONTEXT"))
-    except Exception, e:
-        error("Shotgun: Could not create context! sgtk will be disabled. Details: %s" % e)
-        return
-
-    try:
-        engine = sgtk.platform.start_engine(engine_name, context.tank, context)
-    except Exception, e:
-        error("Shotgun: Could not start engine: %s" % e)
-        return
+    if os.environ.get("SGTK_LOAD_MAX_PLUGINS"):
+        bootstrap_sgtk_with_plugins()
+    else:
+        bootstrap_sgtk_classic()
 
     # if a file was specified, load it now
-    file_to_open =  os.environ.get("TANK_FILE_TO_OPEN")
+    file_to_open = os.environ.get("SGTK_FILE_TO_OPEN")
     if file_to_open:
         MaxPlus.FileManager.Open(file_to_open)
 
     # clean up temp env vars
-    for var in ["TANK_ENGINE", "TANK_CONTEXT", "TANK_FILE_TO_OPEN"]:
+    for var in ["SGTK_ENGINE", "SGTK_CONTEXT", "SGTK_FILE_TO_OPEN", "SGTK_LOAD_MAX_PLUGINS"]:
         if var in os.environ:
             del os.environ[var]
 
```

Comment:
now that this has got multiple callers, i think it's worth pointing out that this is the entry point if you are running from older versions of the multi launch app